src/plan_expand_hypertable.c

```diff
@@ -838,13 +839,18 @@ collect_quals_walker(Node *node, CollectQualCtx *ctx)
 	{
 		FromExpr *f = castNode(FromExpr, node);
 		f->quals = process_quals(f->quals, ctx, false);
-		collect_join_quals(f->quals, ctx, false);
+		/* if this is a nested join we don't propagate join quals */
+		collect_join_quals(f->quals, ctx, ctx->join_level >= 1);
 	}
 	else if (IsA(node, JoinExpr))
 	{
 		JoinExpr *j = castNode(JoinExpr, node);
 		j->quals = process_quals(j->quals, ctx, IS_OUTER_JOIN(j->jointype));
 		collect_join_quals(j->quals, ctx, IS_OUTER_JOIN(j->jointype));
```

Comment:
The parameter name (is_outer_join)  for collect_join_quals should be renamed. It is misleading. Could you rename the parameter to propgate_join_qual or something more appropriate ?  Some comments  about the parameter usage would also be helpful for future reference.